tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/PoPTests.cs

```diff
@@ -451,8 +452,35 @@ public async Task NewPOP_WithKeyIdOnly_Async()
                 result2.AuthenticationResultMetadata.TokenSource);
         }
 
-#if NET_CORE
-        [TestMethod]
+        [RunOn(TargetFrameworks.NetCore)]
+        public async Task WamBadROPCAsync()
+        {
+            var labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+            string[] scopes = { "User.Read" };
+            WamLoggerValidator wastestLogger = new WamLoggerValidator();
+
+            IPublicClientApplication pca = PublicClientApplicationBuilder
+               .Create(labResponse.App.AppId)
+               .WithAuthority(labResponse.Lab.Authority, "organizations")
+               .WithLogging(wastestLogger, enablePiiLogging: true) // it's important that the PII is turned on, otherwise context is 'pii'
+               .WithBroker(new BrokerOptions(BrokerOptions.OperatingSystems.Windows))
+               .Build();
+
+            MsalServiceException ex = await AssertException.TaskThrowsAsync<MsalServiceException>(() =>
+                pca.AcquireTokenByUsernamePassword(
+                    scopes,
+                    "noUser",
+                    "badPassword")                
+                .ExecuteAsync())
+                .ConfigureAwait(false);
+
+            Assert.AreEqual("557973642", ex.AdditionalResponseParameters["RuntimeTag"]);
+            Assert.AreEqual("User name is malformed.", ex.AdditionalResponseParameters["RuntimeContext"]); // message might change. not a big deal
+            Assert.AreEqual("ApiContractViolation", ex.AdditionalResponseParameters["RuntimeStatus"]);
+            Assert.AreEqual("3399811229", ex.AdditionalResponseParameters["RuntimeErrorCode"]);
+        }
```

Comment:
@crmann1 - How about something like this? Not sure if property bag is better suited here as opposed to having new exception hierarchy - `MsalRuntimeServiceException`, `MsalRuntineUiRequiredServiceException` etc. Dedicated exceptions seem a bit overkill.

Reply:
CC @localden for an opinion

Reply:
+1 on the property bug for extensibility of the exception handling and to be consistent across